dotcom-rendering/src/model/enhance-images.ts

```diff
@@ -320,7 +334,7 @@ const addImagePositions = <E extends FEElement>(
 
 		const allImages = isImage(element)
 			? element.media.allImages
-			: element.variants.flatMap((variant) => variant.images);
+			: getCartoonImageForLightbox(element);
 
 		const image = getMaster(allImages) ?? getLargest(allImages);
```

Comment:
Could we consider tweaking this:

```ts
const image = isImage(element)
			? getMaster(element.media.allImages) ?? getLargest(element.media.allImages)
			: getCartoonImageForLightbox(element);
```

and update `getCartoonImageForLightbox` to pick out the specific image we want to use, rather than a list?

```ts
xport const getCartoonImageForLightbox = (
 	element: CartoonBlockElement,
 ): Image | undefined => {
 	const largeVariant = element.variants.find(
 		(variant) => variant.viewportSize === 'large',
 	);
 	return getLargest(largeVariant?.images ?? []); // Or possibly `largeVariant?.images[0]` ?
 };
```

It looks like the representation of images for cartoons is different to that of image elements.

In `ImageBlockElement`, `media.allImages` includes several assets of the _same_ image but at different sizes. It will also, in most cases, include a `master` asset, which is the one we want to use.

By contrast, in `CartoonBlockElement`, `variant.images` includes several _different_ images, all of which are already `master` assets. The one that we want to use is the first (or the largest?) of the images in the `'large'` variant.

Therefore it's not possible to call `getMaster` on cartoon images, because the list of images is a different kind of list to the one that appears on `ImageBlockElement`, and doesn't have a particular "master" asset to be selected.

Reply:
> the first (or the largest?) 
we would want the largest of the images in the 'large' variant yes, but we already do this on the following line:
`const image = getMaster(allImages) ?? getLargest(allImages);`

so that getLargest should continue to work on the list of images returned by this getCartoonImageForLightbox function

but I think you're right in that the variant doesn't containt a list of different sizes for the same image, so in the largest variant you will only ever have one image

let me have a think if I can make it a little clearer what's going on..